In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px
import os

In [2]:
data_path = 'C:\\Users\\Rija\\SHIFTING\\Kaggle-Olympics in Tokyo 2021\\Dataset\\'

athletes_df = pd.read_excel(data_path + 'Athletes.xlsx')
coaches_df = pd.read_excel(data_path + 'Coaches.xlsx')
gender_df = pd.read_excel(data_path + 'EntriesGender.xlsx')
medals_df = pd.read_excel(data_path + 'Medals.xlsx')
teams_df = pd.read_excel(data_path + 'Teams.xlsx')

C:\Users\Rija\Programs\lib\site-packages\openpyxl\styles\stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


In [3]:
print(athletes_df.head(2),'\n')
print(coaches_df.head(2),'\n')
print(gender_df.head(2),'\n')
print(medals_df.head(2),'\n')
print(teams_df.head(2),'\n')

              Name     NOC           Discipline
0  AALERUD Katrine  Norway         Cycling Road
1      ABAD Nestor   Spain  Artistic Gymnastics 

              Name    NOC  Discipline Event
0  ABDELMAGID Wael  Egypt    Football   NaN
1        ABE Junya  Japan  Volleyball   NaN 

       Discipline  Female  Male  Total
0  3x3 Basketball      32    32     64
1         Archery      64    64    128 

   Rank                    Team/NOC  Gold  Silver  Bronze  Total  \
0     1    United States of America    39      41      33    113   
1     2  People's Republic of China    38      32      18     88   

   Rank by Total  
0              1  
1              2   

      Name      Discipline                         NOC Event
0  Belgium  3x3 Basketball                     Belgium   Men
1    China  3x3 Basketball  People's Republic of China   Men 



### Question 1: which countries with the highest participation with Athletes?

In [42]:
athletes_df.describe()

Name                       NOC Discipline
count       11085                     11085      11085
unique      11062                       206         46
top     CHEN Yang  United States of America  Athletics
freq            2                       615       2068

Interpretation: Tokyo Olympics 2021 has 46 Disciplines followed by 206 NOC with a total of 11062 athletes 

In [4]:
c = athletes_df['NOC'].value_counts().sort_values(ascending= False).nlargest(25)

fig=px.bar(x=c.index,y=c.values,
    title="The highest participating Countries",
           labels={"x":"Country","y":"number of Athletes"},
          color_discrete_sequence=["#F98E8A"],
           text=c.values
          )
fig.update_layout(
title_font_color="#F5DCB2",paper_bgcolor="#929494",title_font_size=20,title_x=.5,font_color="#F5DCB2",
    plot_bgcolor="#D0C6AE")
fig.update_yaxes(showgrid=False, color="#F5DCB2")
fig.update_xaxes(tickfont_size=9,color="#F5DCB2")
fig.show()

Interpretation: Countries that have more than 400 athletes are USA, Japan, Australia, ROC, and Germany 

### Question 2: what are the most Discipline in participation?

In [5]:
a=athletes_df["Discipline"].value_counts().sort_values(ascending=False).nlargest(25)

fig=px.bar(x=a.index,y=a.values,title="The highest participating Disciplines",
       labels={"x":"Discipline","y":"number of Athletes"},
        color_discrete_sequence=["#F98E8A"])
fig.update_layout(
title_font_color="#F5DCB2",paper_bgcolor="#929494",title_font_size=20,title_x=.5,font_color="#F5DCB2",
    plot_bgcolor="#D0C6AE")
fig.update_yaxes(showgrid=False,color="#F5DCB2")
fig.update_xaxes(tickfont_size=9,color="#F5DCB2")
fig.show()

Interpretation: the sport with the most athletes is athletics(2068) followed by swimming(743) and football(567) 

### Question 3: Which country produces highest number of coaches?

In [33]:
co = coaches_df['NOC'].value_counts().sort_values().nlargest(25)

fig=px.bar(x=co.index,y=co.values,
    title="The highest participating Countries",
           labels={"x":"Country","y":"number of Athletes"},
          color_discrete_sequence=["#F98E8A"]
          )
fig.update_layout(
title_font_color="#F5DCB2",paper_bgcolor="#929494",title_font_size=20,title_x=.5,font_color="#F5DCB2",
    plot_bgcolor="#D0C6AE")
fig.update_yaxes(showgrid=False, color="#F5DCB2")
fig.update_xaxes(tickfont_size=9,color="#F5DCB2")
fig.update_traces(textfont_size=10)
fig.show()

Interpretasi: At the Tokyo 2021 Olympics, there are three countries with the most contributors of coaches, namely Japan (35), Spain(28), USA(28) 

### Question 4: Gender Across Diciplines

In [37]:
temp_gender = gender_df.sort_values(by= 'Total', ascending= False).loc[:11]
temp_gender

Discipline  Female  Male  Total
4      Athletics     969  1072   2041
36      Swimming     361   418    779
20      Football     264   344    608
29        Rowing     257   265    522
24          Judo     192   201    393
23        Hockey     192   192    384
32      Shooting     178   178    356
31       Sailing     175   175    350
22      Handball     168   168    336
30  Rugby Sevens     146   151    297
45     Wrestling      96   193    289
9         Boxing     102   187    289
42    Volleyball     144   144    288
7     Basketball     144   144    288
43    Water Polo     122   146    268
11  Canoe Sprint     123   126    249

In [43]:

fig = px.bar(gender_df, x='Discipline' , y=['Male','Female'], 
    barmode='group', 
    title="Ten Discipline distribute gender", 
    labels={"x":"Discipline","value":"Number of Athletes"}, 
    color_discrete_sequence=["#CC2A4D", "#F99E4C"])

fig.update_layout(
title_font_color="#F5DCB2",paper_bgcolor="#929494",title_font_size=20,title_x=.5,font_color="#F5DCB2",
    plot_bgcolor="#D0C6AE")
fig.update_yaxes(showgrid=False, color="#F5DCB2")
fig.update_xaxes(tickfont_size=9,color="#F5DCB2")
fig.update_traces(textfont_size=10)
fig.show()

 Interpretation: There is only one discipline with more females than males which is cycling BMX freestyle. But, females participate in all the disciplines whereas men don't in a couple such as rhythmic gymnastics and artistic swimming.

### Question 5: Which country recieved most medals? 

In [24]:
total_medal = medals_df.loc[0:26,:]
final=total_medal.drop(["Rank", "Rank by Total"],axis=1).set_index("Team/NOC").sort_values(by= 'Total', ascending= False)


In [31]:
fig=px.bar(final,x=final.index,y=["Gold","Silver","Bronze"],
       title="The highest countries with total medals",
       labels={"value":"medals","variable":"Medal Type"},
      color_discrete_sequence=["#FFD700","#C0C0C0","#CD7F32"],
       hover_data={"Total":True},
           
      )
fig.update_layout(
title_font_color="#F5DCB2",paper_bgcolor="#929494",title_font_size=20,title_x=.5,font_color="#F5DCB2",
    plot_bgcolor="#D0C6AE",legend_title_font_size=17,legend_title_font_color="#F5DCB2",
    legend_title_text="Medal")
fig.update_yaxes(showgrid=False,title="Total of Medals")
fig.update_xaxes(tickfont_size=9,color="#F5DCB2")
fig.update_traces(textfont_size=10)
fig.show()

interpretation: USA has the highest number of medal winners as well as the country with the most gold medals 